src/bun.js/bindings/bindings.cpp

```diff
@@ -521,6 +525,17 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
             RETURN_IF_EXCEPTION(*scope, false);
         }
 
+        for (; i < array2Length; i++) {
```

Comment:
We are calling the getters for the other array twice?

Reply:
Only if it is longer than the first. We have to check the remaining indexes and make sure they're empty or undefined in non-strict mode.